drivers/video/video.c

```diff
@@ -3200,6 +3198,21 @@ static int video_ioctl(FAR struct file *filep, int cmd, unsigned long arg)
   return ret;
 }
 
+static int video_mmap(FAR struct file *filep, FAR struct mm_map_entry_s *map)
+{
+  FAR struct inode *inode = filep->f_inode;
+  FAR video_mng_t  *priv  = (FAR video_mng_t *)inode->i_private;
+  int ret = -EINVAL;
+
+  if (map)
```

Comment:
let's add after get_bufsize:
```
static size_t get_heapsize(FAR video_type_inf_t *type_inf)
{
  return type_inf->bufinf.container_size *
         get_bufsize(&type_inf->fmt[VIDEO_FMT_MAIN]);
}
```
and do more check here:
```
if (map->offset >= 0 && map->offset < get_heapsize(&priv->video_inf) &&
    map->offset + map->length <= get_heapsize(&priv->video_inf))
```

Reply:
Sorry @xiaoxiang781216, I think you add it while I was hitting the merge button!

Reply:
I will create a patch fix the remain issue.

Reply:
Cool, thanks!

Reply:
Here is the patch: https://github.com/apache/nuttx/pull/8021